src/StatesDumper/StatesDumper.cs

```diff
@@ -122,45 +129,63 @@ private void OnPersistStorage(uint network, DataCache snapshot)
+                _currentBlock = bs_item;
             }
         }
 
+
         private void OnCommitted(NeoSystem system, Block block)
         {
             if (Settings.Default.PersistAction.HasFlag(PersistActions.StorageChanges))
+            {
                 OnCommitStorage(system.Settings.Network, system.StoreView);
+            }
         }
 
         void OnCommitStorage(uint network, DataCache snapshot)
         {
-            if (!bs_cache.TryGetValue(network, out JArray cache)) return;
-            if (cache.Count == 0) return;
+            if (_currentBlock != null)
+            {
+                _writer.WriteLine(_currentBlock.ToString());
+                _writer.Flush();
+            }
+        }
+
+        private void InitFileWriter(uint network, DataCache snapshot)
+        {
             uint blockIndex = NativeContract.Ledger.CurrentIndex(snapshot);
-            if ((blockIndex % Settings.Default.BlockCacheSize == 0) || (Settings.Default.HeightToStartRealTimeSyncing != -1 && blockIndex >= Settings.Default.HeightToStartRealTimeSyncing))
+            if (_writer == null
+                || blockIndex % Settings.Default.BlockCacheSize == 0)
+            {
+                string path = GetOrCreateDirectory(network, blockIndex);
+                var filepart = (blockIndex / Settings.Default.BlockCacheSize) * Settings.Default.BlockCacheSize;
+                path = $"{path}/dump-block-{filepart}.dump";
+                if (_writer != null)
+                {
+                    _writer.Dispose();
+                }
+                _writer = new StreamWriter(new FileStream(path, FileMode.Append));
+            }
+        }
+
+        private string GetOrCreateDirectory(uint network, uint blockIndex)
+        {
+            string dirPathWithBlock = GetDirectoryPath(network, blockIndex);
+            if (_lastCreateDirectory != dirPathWithBlock)
             {
-                string path = HandlePaths(network, blockIndex);
-                path = $"{path}/dump-block-{blockIndex}.json";
-                File.WriteAllText(path, cache.ToString());
-                cache.Clear();
+                Directory.CreateDirectory(dirPathWithBlock);
+                _lastCreateDirectory = dirPathWithBlock;
             }
+            return dirPathWithBlock;
         }
 
-        private static string HandlePaths(uint network, uint blockIndex)
+        private string GetDirectoryPath(uint network, uint blockIndex)
         {
             //Default Parameter
             uint storagePerFolder = 100000;
-            uint folder = (((blockIndex - 1) / storagePerFolder) + 1) * storagePerFolder;
-            if (blockIndex == 0)
-                folder = 0;
-            string dirPathWithBlock = $"./Storage_{network:x8}/BlockStorage_{folder}";
-            Directory.CreateDirectory(dirPathWithBlock);
-            return dirPathWithBlock;
+            uint folder = (blockIndex / storagePerFolder) * storagePerFolder;
```

Comment:
This changes the naming scheme and will break scripts like https://github.com/nspcc-dev/neo-go/blob/master/scripts/compare-dumps/compare-dumps.go. Please keep the old scheme, I doubt changing it is important wrt to this PR goals.

Reply:
Sure, I'll make a new simple plugin so you won't change it.